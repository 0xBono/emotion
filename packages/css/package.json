{
  "name": "@emotion/css",
  "version": "0.9.8",
  "description": "a function to serialize css and object styless",
  "main": "dist/css.cjs.js",
  "module": "dist/css.esm.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/emotion/tree/master/next-packages/css",
  "dependencies": {
    "@emotion/serialize": "^0.9.1",
    "@emotion/utils": "^0.8.2",
    "babel-plugin-emotion": "^9.2.10"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist",
    "macro.js"
  ],
  "umd:main": "./dist/index.min.js",
  "browser": {
    "./dist/css.cjs.js": "./dist/css.browser.cjs.js",
    "./dist/css.esm.js": "./dist/css.browser.esm.js"
  }
}
