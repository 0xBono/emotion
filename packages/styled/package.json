{
  "name": "@emotion/styled",
  "version": "0.10.6",
  "description": "styled API for emotion",
  "main": "dist/styled.cjs.js",
  "module": "dist/styled.esm.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/emotion/tree/master/next-packages/styled",
  "dependencies": {
    "@emotion/styled-base": "^0.10.6",
    "babel-plugin-emotion": "^9.2.10"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist",
    "macro.js"
  ],
  "umd:main": "./dist/index.min.js",
  "browser": {
    "./dist/styled.cjs.js": "./dist/styled.browser.cjs.js",
    "./dist/styled.esm.js": "./dist/styled.browser.esm.js"
  }
}
