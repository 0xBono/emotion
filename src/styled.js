import React from 'react'
import { css } from './index'

export default function (tag, cls, vars = [], content) {
  function Styled (props) {
    const className = css(
      cls,
      vars.map(v => (v && typeof v === 'function' ? v(props) : v)),
      content
    )

    return React.createElement(
      tag,
      Object.assign({}, props, {
        className: props.className
          ? props.className +
          ' ' +
          className
          : className
      })
    )
  }

  Styled.displayName = `Styled[${typeof tag === 'string' ? tag : tag.displayName || 'Wrapped'}](${cls})`

  return Styled
}
