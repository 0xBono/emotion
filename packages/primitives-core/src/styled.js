// @flow
import * as React from 'react'
import {
  channel,
  contextTypes,
  setTheme,
  testAlwaysTrue,
  pickAssign,
  interleave
} from './utils'
import { withCSSContext } from '@emotion/core'
import { createCss } from './css'

let defaultPickTest = prop => prop !== 'theme' && prop !== 'innerRef'

type options = {
  getShouldForwardProp: (cmp: React.ElementType) => (prop: string) => boolean
}

export function createStyled(
  StyleSheet: Object,
  { getShouldForwardProp = () => defaultPickTest }: options = {}
) {
  const css = createCss(StyleSheet)

  return function createEmotion(component: React.ElementType) {
    let pickTest = getShouldForwardProp(component)

    return function createStyledComponent(...rawStyles: *) {
      let styles

      if (rawStyles[0] == null || rawStyles[0].raw === undefined) {
        styles = rawStyles
      } else {
        styles = interleave(rawStyles)
      }

      // do we really want to use the same infra as the web since it only really uses theming?
      let Styled = withCSSContext((props, context) => {
        let mergedProps = pickAssign(testAlwaysTrue, {}, props, {
          theme: props.theme || context.theme
        })
        let stylesWithStyleProp = styles
        if (props.style) {
          stylesWithStyleProp = styles.concat(props.style)
        }
        const emotionStyles = css.apply(mergedProps, stylesWithStyleProp)

        return React.createElement(
          component,
          pickAssign(pickTest, {}, props, {
            ref: props.innerRef,
            style: emotionStyles
          })
        )
      })
      // $FlowFixMe
      Styled.withComponent = (newComponent: React.ElementType) =>
        createEmotion(newComponent)(...styles)

      Styled.displayName = `emotion(${getDisplayName(component)})`

      return Styled
    }
  }
}

const getDisplayName = primitive =>
  typeof primitive === 'string'
    ? primitive
    : primitive.displayName || primitive.name || 'Styled'
