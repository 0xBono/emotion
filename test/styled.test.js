/* eslint-disable jsx-quotes,no-useless-escape,no-template-curly-in-string */
/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
// eslint-disable-next-line no-unused-vars
import css, { fragment } from '../lib/index'
import { styled } from '../lib/styled'

describe('glam react', () => {
  test('basic render', () => {
    const fontSize = 20
    const H1 = styled.h1`
      font-size: ${fontSize}px;
    `

    const tree = renderer
      .create(
        <H1>
          hello world
        </H1>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('attr', () => {
    const H1 = styled.h1`
      font-size: attr(fontSize);
      margin: attr(margin rem, 4);
    `

    const Title = ({ title }) => {
      return (
        <H1 fontSize={48}>
          {title}
        </H1>
      )
    }

    const tree = renderer.create(<Title />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('call expression', () => {
    const fontSize = 20
    const H1 = styled('h1')`
      font-size: ${fontSize}px;
    `

    const tree = renderer
      .create(
        <H1 className={'legacy__class'}>
          hello world
        </H1>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('composition', () => {
    const fontSize = 20
    const H1 = styled('h1')`
      font-size: ${fontSize}px;
    `

    const H2 = styled(H1)`font-size: ${fontSize * 2 / 3}`

    const tree = renderer
      .create(
        <H2 className={'legacy__class'}>
          hello world
        </H2>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('function in expression', () => {
    const fontSize = 20
    const H1 = styled('h1')`
      font-size: ${fontSize}px;
    `

    const H2 = styled(H1)`font-size: ${({ scale }) => fontSize * scale}`

    const tree = renderer
      .create(
        <H2 scale={2} className={'legacy__class'}>
          hello world
        </H2>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('fragments', () => {
    const fontSize = 20

    const fragA = fragment`
      color: ${({ color }) => color};
    `

    const fragB = fragment`
      height: ${({ height }) => height}px;
      @apply ${fragA}
    `

    const H1 = styled('h1')`
      font-size: ${fontSize}px;
      @apply ${fragB}
    `

    const H2 = styled(H1)`font-size: ${({ scale }) => fontSize * scale}`

    const tree = renderer
      .create(
        <H2 scale={2} className={'legacy__class'}>
          hello world
        </H2>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('higher order component', () => {
    const fontSize = 20
    const Content = styled('div')`
      font-size: ${fontSize}px;
    `

    const flexColumn = Component => {
      const NewComponent = styled(Component)`
        flex-direction: column;
      `
      NewComponent.displayName = `flexColumn${Component.displayName}`

      return NewComponent
    }

    const ColumnContent = flexColumn(Content)

    expect(ColumnContent.displayName).toBe('flexColumnundefined')

    const tree = renderer.create(<ColumnContent />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
