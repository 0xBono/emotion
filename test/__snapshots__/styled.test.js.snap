// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glam react attr 1`] = `
<h1
  className="css-15vt02v-1mu1h6y"
  fontSize={48}
/>
`;

exports[`glam react basic render 1`] = `
<h1
  className="css-13wdnau-1ejkrg"
>
  hello world
</h1>
`;

exports[`glam react call expression 1`] = `
<h1
  className="legacy__class css-13wdnau-1ejkrg"
>
  hello world
</h1>
`;

exports[`glam react composition 1`] = `
<h1
  className="legacy__class css-vxb7tq-tlgws4 css-13wdnau-1ejkrg"
>
  hello world
</h1>
`;

exports[`glam react fragments 1`] = `
<h1
  className="legacy__class css-vxb7tq-1qdhtcv css-91i0rz-xcpsbd"
  scale={2}
>
  hello world
</h1>
`;

exports[`glam react function in expression 1`] = `
<h1
  className="legacy__class css-vxb7tq-1qdhtcv css-13wdnau-1ejkrg"
  scale={2}
>
  hello world
</h1>
`;

exports[`glam react higher order component 1`] = `
<div
  className="css-o7j1h0 css-13wdnau-1ejkrg"
/>
`;
